Detectors/GlobalTracking/include/GlobalTracking/MatchHMP.h

```diff
@@ -109,9 +109,9 @@ class MatchHMP
                             ITSTPCTRD,
                             SIZEALL };
 
-  std::vector<o2::dataformats::MatchInfoHMP>& getMatchedTrackVector(o2::globaltracking::MatchHMP::trackType index) { return mMatchedTracks[index]; }
+  std::vector<o2::dataformats::MatchInfoHMP> getMatchedTrackVector(o2::globaltracking::MatchHMP::trackType index) { return mMatchedTracks[index]; }
 
-  std::vector<o2::MCCompLabel>& getMatchedHMPLabelsVector(o2::globaltracking::MatchHMP::trackType index) { return mOutHMPLabels[index]; } ///< get vector of HMP label of matched tracks
+  std::vector<o2::MCCompLabel> getMatchedHMPLabelsVector(o2::globaltracking::MatchHMP::trackType index) { return mOutHMPLabels[index]; } ///< get vector of HMP label of matched tracks
```

Comment:
Why did you change the returned type from the reference to copy?

Reply:
Hi Ruben, because when I changed the MatchInfo pointer object to stack object as you suggested, the wokflow failed and like this not!

Reply:
getMatchedTrackVector is used only to pass the vector to DPL snapchot, I don't see how it can be related to changing the pointer by object well before that. What was the failure?

Reply:
It is was strange also for me! Anyway, now I retried again, and it is working, maybe there was an other reason! I have just committed the revised code.